backend/transcript/views.py

```diff
@@ -383,12 +390,31 @@ def save_transcription(request):
     except Transcript.DoesNotExist:
 
         # Collect the video object
-        video_id = request.data["video_id"]
-        video = Video.objects.get(pk=video_id)
+        try: 
+            video_id = request.data["video_id"]
+            video = Video.objects.get(pk=video_id)
+        except KeyError:
+            return Response(
+                {"message": "Missing required parameters - video_id or video does not exist."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Check if a transcript exists for this video, language, user and transcript_type
+        transcript = Transcript.objects.filter(
+            video=video, language=language, user=user_id, transcript_type=MANUALLY_CREATED
+        ).first() 
+        
+        # If a transcript exists, update the payload
+        if transcript:
+            transcript.payload = transcribed_data
+            transcript.save()
+
+            return Response(
+                {"id": transcript.id, "data": transcript.payload},
+                status=status.HTTP_200_OK,
+            )
```

Comment:
Just a small refactoring suggestion, shall we make it like this?

```
if transcript:
    transcript.payload = transcribed_data
else:
    transcript = # new transcript
transcript.save()
return response
```

Reply:
yeah sure

Reply:
Done